### ISSUE TITLE
Custom model not working

### ISSUE THREAD

**Issue by timebreak** [home-assistant/addons#3331]
### Describe the issue you are experiencing

I have custom model in "/share/piper" with both .onnx and onnx.json and keep getting this error when trying to generate audio file in HA:

```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service piper: starting
s6-rc: info: service piper successfully started
s6-rc: info: service discovery: starting
DEBUG:wyoming_piper.download:Downloading https://huggingface.co/rhasspy/piper-voices/resolve/v1.0.0/voices.json to /data/voices.json
DEBUG:wyoming_piper.download:Loading /data/voices.json
DEBUG:wyoming_piper.process:Starting process for: en_US-lessac-medium (1/1)
DEBUG:wyoming_piper.download:Checking /data/en_US-lessac-medium.onnx
DEBUG:wyoming_piper.download:Checking /data/en_US-lessac-medium.onnx.json
DEBUG:wyoming_piper.process:Starting piper process: /usr/share/piper/piper args=['--model', '/data/en_US-lessac-medium.onnx', '--config', '/data/en_US-lessac-medium.onnx.json', '--output_dir', '/tmp/tmpunycmmoe', '--json-input', '--noise-scale', '0.667', '--length-scale', '1.0', '--noise-w', '0.2']
INFO:__main__:Ready
[19:14:26] INFO: Successfully send discovery information to Home Assistant.
s6-rc: info: service discovery successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
DEBUG:wyoming_piper.handler:Sent info
DEBUG:wyoming_piper.handler:Sent info
DEBUG:wyoming_piper.handler:Sent info
DEBUG:wyoming_piper.handler:Synthesize(text='Hello there, still you?', voice=SynthesizeVoice(name='linhi', language=None, speaker=None))
DEBUG:wyoming_piper.handler:synthesize: raw_text=Hello there, still you?, text='Hello there, still you?'
DEBUG:wyoming_piper.process:Stopping process for: en_US-lessac-medium
DEBUG:wyoming_piper.process:Starting process for: linhi (1/1)
ERROR:asyncio:Task exception was never retrieved
future: <Task finished name='Task-18' coro=<AsyncEventHandler.run() done, defined at /usr/local/lib/python3.11/dist-packages/wyoming/server.py:28> exception=VoiceNotFoundError('linhi')>
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/dist-packages/wyoming/server.py", line 35, in run
    if not (await self.handle_event(event)):
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/wyoming_piper/handler.py", line 73, in handle_event
    piper_proc = await self.process_manager.get_process(voice_name=voice_name)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/wyoming_piper/process.py", line 114, in get_process
    ensure_voice_exists(
  File "/usr/local/lib/python3.11/dist-packages/wyoming_piper/download.py", line 77, in ensure_voice_exists
    find_voice(name, data_dirs)
  File "/usr/local/lib/python3.11/dist-packages/wyoming_piper/download.py", line 183, in find_voice
    raise VoiceNotFoundError(name)
wyoming_piper.download.VoiceNotFoundError: linhi
```

model files:  en_US-linhi-piper_train.onnx.json and en_US-linhi-piper_train.onnx

json data:
```
{
    "dataset": "linhi",
    "audio": {
        "sample_rate": 22050,
        "quality": "piper_train"
    },
    "espeak": {
        "voice": "en-us"
    },
    "language": {
		"code": "en_US",
		"family": "en",
		"region": "US",
		"name_native": "English",
		"name_english": "English",
		"country_english": "United States"
    },
    "inference": {
        "noise_scale": 0.667,
        "length_scale": 1,
        "noise_w": 0.8
    },
    "phoneme_type": "espeak",
    "phoneme_map": {},
    "phoneme_id_map": {
      [MAP HERE (I don't copy)]
    },
    "num_symbols": 256,
    "num_speakers": 1,
    "speaker_id_map": {},
    "piper_version": "1.0.0"
}
```

model does work in virtual machine where I trained it

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Which add-on are you reporting an issue with?

Piper

### What is the version of the add-on?

1.4.0

### Steps to reproduce the issue

1. Follow Piper add-on description for using custom model
2. Use it In HA
3. Try generate audio with it

### System Health information

None

**Comment by haroldboom** [comment]
Hi,
I am having the same issue.

DEBUG:wyoming_piper.handler:Sent info
DEBUG:wyoming_piper.handler:Synthesize(text='Hello. How can I assist?', voice=SynthesizeVoice(name='piper', language=None, speaker=None))
DEBUG:wyoming_piper.handler:synthesize: raw_text=Hello. How can I assist?, text='Hello. How can I assist?'
DEBUG:wyoming_piper.process:Stopping process for: en_GB-alan-low
DEBUG:wyoming_piper.process:Starting process for: piper (1/1)
ERROR:asyncio:Task exception was never retrieved
future: <Task finished name='Task-12' coro=<AsyncEventHandler.run() done, defined at /usr/local/lib/python3.11/dist-packages/wyoming/server.py:28> exception=VoiceNotFoundError('piper')>
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/dist-packages/wyoming/server.py", line 35, in run
    if not (await self.handle_event(event)):
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/wyoming_piper/handler.py", line 73, in handle_event
    piper_proc = await self.process_manager.get_process(voice_name=voice_name)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/wyoming_piper/process.py", line 114, in get_process
    ensure_voice_exists(
  File "/usr/local/lib/python3.11/dist-packages/wyoming_piper/download.py", line 77, in ensure_voice_exists
    find_voice(name, data_dirs)
  File "/usr/local/lib/python3.11/dist-packages/wyoming_piper/download.py", line 183, in find_voice
    raise VoiceNotFoundError(name)
wyoming_piper.download.VoiceNotFoundError: piper


json file;
{
    "dataset": "piper",
    "audio": {
        "sample_rate": 22050,
        "quality": "hal9000_training"
    },
    "espeak": {
        "voice": "en"
    },
    "language": {
        "code": "en"
    },
    "inference": {
        "noise_scale": 0.667,
        "length_scale": 1,
        "noise_w": 0.8
    },
    "phoneme_type": "espeak",
    "phoneme_map": {},
    "phoneme_id_map": {
        " ": [ map removed
        
    "num_symbols": 256,
    "num_speakers": 1,
    "speaker_id_map": {},
    "piper_version": "1.0.0"
}

**Comment by alealexpro100** [comment]
Try to rename model files to `<model_name>.onnx` template.
For example, if model's name is `linhi` (it is shown in exception from logs: `wyoming_piper.download.VoiceNotFoundError: linhi`) then its files are `linhi.onnx` and `linhi.onnx.json`.

**Comment by haroldboom** [comment]
Hey mate, yep that worked! Thanks

**Comment by timebreak** [comment]
@alealexpro100 
Sure! Can't believe I did not try that.
Thank you

**Comment by mylegitches** [comment]
Jumping in here...  I have renamed my files to match the dataset value in my .json. 
I have the .json:
```
{
    "dataset": "voice",
    "audio": {
        "sample_rate": 22050,
        "quality": "my-training"
    },
    "espeak": {
        "voice": "en"
    },
    "language": {
        "code": "en"
    },
```
I have added my model files (using samba plugin) to \\homeassistant\share\piper
\\homeassistant\share\piper\voice.onnx
\\homeassistant\share\piper\voice.onnx.json

I have restarted piper, HA, and the rpi.  Still, I am not able to see it in the list.


@haroldboom - I want that Hal9000 model!  I am working on a Chucky (Charles Lee Ray voice) model now LOL. Will trade?
